src/System.Management.Automation/FormatAndOutput/common/ColumnWidthManager.cs

```diff
@@ -107,18 +115,18 @@ private bool AssignColumnWidths(Span<int> columnWidths)
             }
 
             // we still have room and we want to add more width
-
-            while (availableWidth > 0)
+            if (unfixedColumnCount > 0)
             {
+                int extraSpace = availableWidth / unfixedColumnCount;
+                int remainder = availableWidth % unfixedColumnCount;
+
                 for (int k = 0; k < columnWidths.Length; k++)
                 {
-                    if (fixedColumn[k])
-                        continue;
-
-                    columnWidths[k]++;
-                    availableWidth--;
-                    if (availableWidth == 0)
-                        break;
+                    if (!fixedColumn[k])
+                    {
+                        columnWidths[k] += extraSpace;
+                        if (remainder-- > 0) columnWidths[k]++;
```

Comment:
I believe there is an off by one bug here, where if for example the unfixedColumnCount is 4 and the remainder is 3, the one extra space will only be applied to two columns. Documenting it now so I can fix it when I can get back to this PR.

Reply:
It would be nice to add new tests to cover the code.

Reply:
I have added some unit tests in my latest commit. I updated Format-Table.Tests.ps1.